src/clj_momo/lib/clj_time/core.cljc

```diff
@@ -0,0 +1,614 @@
+     (after? [this ^ReadableInstant that]
+       (time-delegate/after? this that))
+     (before? [this ^ReadableInstant that]
+       (time-delegate/before? this that))
+     (plus- [this ^ReadableInstant period]
+       (time-delegate/plus- this period))
+     (minus- [this ^ReadableInstant period]
+       (time-delegate/minus- this period))
+     (first-day-of-the-month- [this]
+       (time-delegate/first-day-of-the-month- this))
+     (last-day-of-the-month- [this]
+       (time-delegate/last-day-of-the-month- this))
+     (week-number-of-year [this]
+       (time-delegate/week-number-of-year this))
+
+     org.joda.time.LocalDate
+     (year [this] (time-delegate/year this))
+     (month [this] (time-delegate/month this))
+     (day [this] (time-delegate/day this))
+     (day-of-week [this] (time-delegate/day-of-week this))
+     (hour [this] (time-delegate/hour this))
+     (minute [this] (time-delegate/minute this))
+     (second [this] (time-delegate/second this))
+     (milli [this] (time-delegate/milli this))
+     (equal? [this ^ReadableInstant that]
+       (time-delegate/equal? this that))
+     (after? [this ^ReadableInstant that]
+       (time-delegate/after? this that))
+     (before? [this ^ReadableInstant that]
+       (time-delegate/before? this that))
+     (plus- [this ^ReadableInstant period]
+       (time-delegate/plus- this period))
+     (minus- [this ^ReadableInstant period]
+       (time-delegate/minus- this period))
+     (first-day-of-the-month- [this]
+       (time-delegate/first-day-of-the-month- this))
+     (last-day-of-the-month- [this]
+       (time-delegate/last-day-of-the-month- this))
+     (week-number-of-year [this]
+       (time-delegate/week-number-of-year this))
+
+     org.joda.time.YearMonth
+     (year [this] (time-delegate/year this))
+     (month [this] (time-delegate/month this))
+     (equal? [this ^ReadablePartial that]
+       (time-delegate/equal? this that))
+     (after? [this ^ReadablePartial that]
+       (time-delegate/after? this that))
+     (before? [this ^ReadablePartial that]
+       (time-delegate/before? this that))
+     (plus- [this ^ReadablePeriod period]
+       (time-delegate/plus- this period))
+     (minus- [this ^ReadablePeriod period]
+       (time-delegate/minus- this period))
+
+     org.joda.time.LocalTime
+     (hour [this] (time-delegate/hour this))
+     (minute [this] (time-delegate/minute this))
+     (second [this] (time-delegate/second this))
+     (milli [this] (time-delegate/milli this))
+     (equal? [this ^ReadablePartial that]
+       (time-delegate/equal? this that))
+     (after? [this ^ReadablePartial that]
+       (time-delegate/after? this that))
+     (before? [this ^ReadablePartial that]
+       (time-delegate/before? this that))
+     (plus- [this ^ReadablePeriod period]
+       (time-delegate/plus- this period))
+     (minus- [this ^ReadablePeriod period]
+       (time-delegate/minus- this period))
+
+     java.util.Date
```

Comment:
These all contain a repeated pattern. What about factoring out to a helper function?
`(defn time-get [^java.util.Date date field]
  (.get (doto (GregorianCalendar.) (.setTimeZone TimeZoneUTC) (.setTime date)) field))`
So the `year` function, for instance, becomes:
`(year [this] (time-get Calendar/YEAR))`